### ISSUE TITLE
LDAP password change should rebind as user 

### ISSUE THREAD

**Issue by politas** [nextcloud/server#25054]
**How to support this request?**
- Please use the 👍  [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are also in favor of this feature - we are using this information to decide which features to include in upcoming Nextcloud versions.
- Please don't comment if you have no relevant information to add

---

**Password changing for LDAP authenticated users require server settings to have write capable user**
Usual LDAP system authentication systems only require a read-only user to search for user accounts. Putting a read-only account in the system LDAP settings does not allow users to change their LDAP passwords. The server tries to use the read-only account to change LDAP, which fails, even when the LDAP olcAccess is set to allow users to change their own password as is usually described as a requirement for LDAP password changes to be permitted.

**Describe the solution you'd like**
For password changes, NextCloud should rebind using the user's authentication in order to write the changes, allowing limited access to be given to the NextCloud server.

**Comment by politas** [comment]
I'll point out that doing password changes using an admin account is contrary to, and overrides, the LDAP olcAccess standard for allowing own password changes. NextCloud should not give users more access to a connected LDAP system than that system is configured to give.

**Comment by Caligatio** [comment]
I just switched my Nextcloud server over to using LDAP and this issue probably cost me about 90 minutes of troubleshooting to figure out. My LDAP server allows for anonymous authentication so I have not needed to give any applications accounts thus far.

Rebinding as the user that wants to change their password would be ideal.

**Comment by szaimen** [comment]
cc @nextcloud/ldap is this feasible?

**Comment by politas** [comment]
Any pointers as to what the "missing info" might be?

**Comment by Caligatio** [comment]
Bumping this as it's not stale.

**Comment by politas** [comment]
Still wondering what the missing info is. Please advise what more is needed to stop this getting marked as stale over and over again. Not feeling a lot of love for LDAP authentication issues from Nextcloud devs.

**Comment by Eiskasten** [comment]
I fully agree on that. In addition, password resets should remain to rebind as the configured administrator in order to let users reset their password without providing their old one.

**Comment by Caligatio** [comment]
Bumping to avoid this issue being closed

**Comment by Eiskasten** [comment]
For everybody who requires this feature: I switched to [Self Service Password](https://github.com/ltb-project/self-service-password) which behaves as I would expect from a password reset functionality using an LDAP. The major drawback is that I have to host an additional web application now. This lead me to completely disable the password management functions in Nextcloud.

The only way I think of to switch back to the Nextcloud password management is when:

- for user password resets which include their password, Nextcloud uses the user account for the reset on the LDAP
- for password resets without the user password, an admin user from the LDAP will be used
- Nextcloud (optionally) enforces a password policy (given by its dn) manually to all password resets triggered by the admin user

By the way: Nextcloud already has the ability to specify a dn for a password policy. But for whatever reason they do not seem to be enforced. But this is a story for another ticket.

**Comment by Caligatio** [comment]
Bumping to avoid this issue being closed

**Comment by Caligatio** [comment]
Still not stale and still not missing any information.

**Comment by Caligatio** [comment]
Bump

**Comment by politas** [comment]
Can we get someone from @nextcloud/ldap to at least comment on this?

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by politas** [comment]
Still waiting for developers to notice this issue and explain any required further information.

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by politas** [comment]
Still waiting for developers to notice this issue and explain any required further information.

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by politas** [comment]
Still waiting for developers to notice this issue and explain any required further information.

**Comment by politas** [comment]
@szaimen, could you perhaps remove the "needs info" tag on this issue so it doesn't require regular bumping?

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by politas** [comment]
@szaimen, could you perhaps remove the "needs info" tag on this issue so it doesn't require regular bumping?
